cli/src/cli_util.rs

```diff
@@ -941,6 +941,19 @@ impl WorkspaceCommandHelper {
         self.workspace.workspace_id()
     }
 
+    pub fn workspace_contains_actual_repo(&self) -> bool {
+        // we won't assume your default workspace is called "default"
+        //
+        // repo_path is not canonicalized (and may be relative) whereas workspace_root is.
+        self.repo_path()
+            .canonicalize()
```

Comment:
Instead of trying to canonicalize paths for comparing them, wouldn't it suffice to check whether the current workspace as a `.jj/repo` directory as its root? Something like:

```rust
    fs::metadata(self.workspace_root().join(".jj").join("repo")).is_ok_and(|s| s.is_dir())
```

Reply:
Ah, this commit was intended to be be landed separately as it will need a bit of chat about when it's okay to delete the default workspace, and whether we can implement turning the repo back from a bare repo.